### DISCUSSION TITLE
"Nested" environment not showing in prompt

### DISCUSSION THREAD

**Discussion by RankkaApina** [JanDeDobbeleer/oh-my-posh#4174] [Troubleshoot]
I searched and haven't found an answer, it's possible that I'm trying to do something that won't work. I use Pyenv to configure my python, and my prompt looks nice and I see which python I'm using. However, when I make a conda environment (conda installed from pyenv), I do see the correct python if it's different from the default in the folder, but the environment doesn't change.

So, when the environment is not activated, this is what the prompt shows:
3.10.12 (miniforge3-latest)
When I activate my conda environment (called geo), I see:
3.11.4 (miniforge3-latest)

I get that this is because oh-my-posh is seeing the pyenv as the virtual environment, but is there a way to change it to the conda or alternatively add the nested environment to the prompt?

**Comment by JanDeDobbeleer** [comment]
@RankkaApina in this case, can you run oh-my-posh debug and show the logs for the Python segment? It's going to be the environment variables who're providing this info, I'd love to see which one.

**Reply by RankkaApina** [reply]
Here's the debug. I am also wondering if this has something to do with the fact that I'm using WSL. I had issues with git branches showing up before I installed git-bash on the windows side. Let me know if you want any other info from me :)


Version: 16.4.0

Shell: zsh (5.8.1)

Segments:

ConsoleTitle(true) -   0 ms - Ubuntu-22.04
session(true)      -   0 ms - pingu
path(true)         -   1 ms -    Projects  geoprojects
python(true)       -   2 ms -   3.11.4 (miniforge3-latest)
git(true)          -   2 ms -   (main) ♥
time(true)         -   0 ms -  18:04:11

Run duration: 7.722834ms

Cache path: /home/pingu/.cache/oh-my-posh

Config path: /home/pingu/proj/dotfiles/zsh/poshthemes/0_nord-home.omp.json

Logs:
[DEBUG] 17:38:51.802 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.802 shell.go:Getenv(TERM_PROGRAM) - 89.89µs
[DEBUG] 17:38:51.802 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.802 shell.go:Getenv(XDG_CACHE_HOME) - 7.073µs
[TRACE] 17:38:51.802 shell_unix.go:CachePath() - 16.181µs
[DEBUG] 17:38:51.802 shell.go:Getenv:395 → /home/pingu/proj/dotfiles/zsh/poshthemes/0_nord-home.omp.json
[TRACE] 17:38:51.802 shell.go:Getenv(POSH_THEME) - 4.027µs
[DEBUG] 17:38:51.802 shell_unix.go:Platform:71 → ubuntu
[TRACE] 17:38:51.802 shell.go:resolveConfigPath() - 10.7µs
[TRACE] 17:38:51.802 shell.go:Init() - 301.203µs
[TRACE] 17:38:51.802 shell.go:Flags() - 41ns
[TRACE] 17:38:51.802 config.go:loadConfig() - 284.13µs
[TRACE] 17:38:51.802 shell.go:Flags() - 40ns
[DEBUG] 17:38:51.802 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.802 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.164µs
[DEBUG] 17:38:51.803 shell.go:Shell:642 → process name: zsh
[TRACE] 17:38:51.803 shell.go:Shell() - 489.45µs
[DEBUG] 17:38:51.803 shell.go:Getenv:395 → 5.8.1
[TRACE] 17:38:51.803 shell.go:Getenv(POSH_SHELL_VERSION) - 2.285µs
[DEBUG] 17:38:51.803 debug.go:PrintDebug:22 → Segment: Title
[TRACE] 17:38:51.803 shell.go:Flags() - 40ns
[DEBUG] 17:38:51.803 block.go:Debug:230 → Segment: session
[TRACE] 17:38:51.803 shell.go:GOOS() - 40ns
[DEBUG] 17:38:51.803 shell.go:Pwd:425 → /home/pingu/proj/geoprojects
[TRACE] 17:38:51.803 shell.go:Pwd() - 6.913µs
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[DEBUG] 17:38:51.803 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.803 shell.go:Getenv(SSH_CONNECTION) - 1.794µs
[DEBUG] 17:38:51.803 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.803 shell.go:Getenv(SSH_CLIENT) - 1.513µs
[TRACE] 17:38:51.803 shell_unix.go:Root() - 170ns
[TRACE] 17:38:51.803 shell.go:Shell() - 60ns
[TRACE] 17:38:51.803 shell.go:ErrorCode() - 40ns
[DEBUG] 17:38:51.803 shell_unix.go:IsWsl:35 → true
[TRACE] 17:38:51.803 shell_unix.go:IsWsl() - 2.555µs
[TRACE] 17:38:51.803 shell.go:Pwd() - 70ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 21ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[DEBUG] 17:38:51.803 shell.go:User:546 → pingu
[TRACE] 17:38:51.803 shell.go:User() - 2.074µs
[DEBUG] 17:38:51.803 shell.go:Host:558 → anna
[TRACE] 17:38:51.803 shell.go:Host() - 2.775µs
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[DEBUG] 17:38:51.803 shell_unix.go:Platform:71 → ubuntu
[DEBUG] 17:38:51.803 shell.go:Getenv:395 → 1
[TRACE] 17:38:51.803 shell.go:Getenv(SHLVL) - 1.764µs
[TRACE] 17:38:51.803 shell.go:TemplateCache() - 83.068µs
[TRACE] 17:38:51.803 shell.go:TemplateCache() - 80ns
[TRACE] 17:38:51.803 shell.go:Shell() - 90ns
[DEBUG] 17:38:51.803 block.go:Debug:230 → Segment: path
[TRACE] 17:38:51.803 shell.go:Pwd() - 50ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:Pwd() - 30ns
[TRACE] 17:38:51.803 shell.go:Shell() - 41ns
[TRACE] 17:38:51.803 shell.go:Shell() - 20ns
[DEBUG] 17:38:51.803 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 17:38:51.803 properties.go:GetString:28 → 
[DEBUG] 17:38:51.803 properties.go:GetString:28 → 
[TRACE] 17:38:51.803 shell.go:GOOS() - 80ns
[DEBUG] 17:38:51.803 properties.go:GetString:28 → 
[DEBUG] 17:38:51.803 properties.go:GetKeyValueMap:46 → mapped_locations: map[/home/pingu/proj:  Projects /home/pingu/proj/vault/memory-alpha:  Memory Alpha /home/pingu/proj/vault/memory-beta-crafts:  Memory Beta Crafts /home/pingu/proj/vault/memory-beta-languages:  Memory Beta Languages /home/pingu/proj/vault/memory-beta-space:  Memory Beta Space /home/pingu/proj/vault/memory-beta-tech:  Memory Beta Tech]
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 40ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 21ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 20ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.803 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[DEBUG] 17:38:51.803 properties.go:GetString:28 → agnoster_full
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[DEBUG] 17:38:51.803 properties.go:GetStringArray:52 → cycle: []
[DEBUG] 17:38:51.803 properties.go:GetString:28 → NO DATA
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[DEBUG] 17:38:51.803 properties.go:GetString:28 →  
[DEBUG] 17:38:51.803 properties.go:GetBool:22 → cycle_folder_separator: false
[TRACE] 17:38:51.803 shell.go:PathSeparator() - 30ns
[TRACE] 17:38:51.803 shell.go:Pwd() - 50ns
[DEBUG] 17:38:51.803 shell_unix.go:IsWsl:35 → true
[TRACE] 17:38:51.803 shell_unix.go:IsWsl() - 2.014µs
[DEBUG] 17:38:51.805 shell.go:RunCommand:576 → //wsl.localhost/Ubuntu-22.04/home/pingu/proj/geoprojects
[TRACE] 17:38:51.805 shell.go:RunCommand(wslpath -m /home/pingu/proj/geoprojects) - 1.960766ms
[TRACE] 17:38:51.805 shell.go:StackCount() - 150ns
[TRACE] 17:38:51.805 shell_unix.go:DirIsWritable(/home/pingu/proj/geoprojects) - 3.897µs
[TRACE] 17:38:51.805 shell.go:TemplateCache() - 70ns
[TRACE] 17:38:51.805 shell.go:TemplateCache() - 80ns
[TRACE] 17:38:51.806 shell.go:Shell() - 100ns
[DEBUG] 17:38:51.806 block.go:Debug:230 → Segment: python
[DEBUG] 17:38:51.806 properties.go:GetString:28 → always
[TRACE] 17:38:51.806 shell.go:Pwd() - 80ns
[TRACE] 17:38:51.806 shell.go:GOOS() - 40ns
[DEBUG] 17:38:51.806 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 17:38:51.806 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 17:38:51.807 properties.go:GetBool:22 → home_enabled: false
[TRACE] 17:38:51.807 shell.go:Pwd() - 190ns
[DEBUG] 17:38:51.807 properties.go:GetBool:22 → fetch_virtual_env: true
[DEBUG] 17:38:51.807 shell.go:CommandPath:597 → /home/pingu/.pyenv/versions/miniforge3-latest/envs/geo/bin/python
[TRACE] 17:38:51.807 shell.go:CommandPath(python) - 27.452µs
[DEBUG] 17:38:51.807 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.807 shell.go:HasFilesInDir(pyvenv.cfg) - 32.792µs
[DEBUG] 17:38:51.807 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.807 shell.go:HasFilesInDir(pyvenv.cfg) - 9.377µs
[DEBUG] 17:38:51.807 shell.go:Getenv:395 → /home/pingu/.pyenv/versions/miniforge3-latest
[TRACE] 17:38:51.807 shell.go:Getenv(VIRTUAL_ENV) - 4.899µs
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[TRACE] 17:38:51.807 shell.go:GOOS() - 20ns
[DEBUG] 17:38:51.807 properties.go:GetBool:22 → display_default: true
[DEBUG] 17:38:51.807 properties.go:GetBool:22 → fetch_version: true
[DEBUG] 17:38:51.807 shell.go:CommandPath:591 → /home/pingu/.pyenv/versions/miniforge3-latest/envs/geo/bin/python
[TRACE] 17:38:51.807 shell.go:CommandPath(python) - 2.545µs
[DEBUG] 17:38:51.807 shell.go:Getenv:395 → /home/pingu/.pyenv
[TRACE] 17:38:51.807 shell.go:Getenv(PYENV_ROOT) - 3.637µs
[DEBUG] 17:38:51.807 shell.go:CommandPath:591 → /home/pingu/.pyenv/versions/miniforge3-latest/envs/geo/bin/python
[TRACE] 17:38:51.807 shell.go:CommandPath(python) - 2.205µs
[TRACE] 17:38:51.807 shell.go:HasCommand(python) - 4.158µs
[DEBUG] 17:38:51.810 shell.go:RunCommand:576 → Python 3.11.4
[TRACE] 17:38:51.810 shell.go:RunCommand(python --version) - 3.017602ms
[DEBUG] 17:38:51.810 properties.go:GetString:28 → https://docs.python.org/release/{{ .Major }}.{{ .Minor }}.{{ .Patch }}/whatsnew/changelog.html#python-{{ .Major }}-{{ .Minor }}-{{ .Patch }}
[TRACE] 17:38:51.810 shell.go:TemplateCache() - 100ns
[TRACE] 17:38:51.810 shell.go:TemplateCache() - 60ns
[TRACE] 17:38:51.810 shell.go:TemplateCache() - 90ns
[TRACE] 17:38:51.810 shell.go:Shell() - 130ns
[DEBUG] 17:38:51.810 block.go:Debug:230 → Segment: git
[TRACE] 17:38:51.810 shell.go:Pwd() - 80ns
[TRACE] 17:38:51.810 shell.go:GOOS() - 40ns
[DEBUG] 17:38:51.810 shell_unix.go:IsWsl:35 → true
[TRACE] 17:38:51.810 shell_unix.go:IsWsl() - 3.036µs
[DEBUG] 17:38:51.810 shell.go:FileContent:520 ↓
    5.15.90.1-microsoft-standard-WSL2

[TRACE] 17:38:51.810 shell.go:FileContent(/proc/sys/kernel/osrelease) - 39.435µs
[TRACE] 17:38:51.810 shell_unix.go:IsWsl2() - 47.931µs
[TRACE] 17:38:51.810 shell.go:Pwd() - 60ns
[DEBUG] 17:38:51.812 shell.go:RunCommand:576 → //wsl.localhost/Ubuntu-22.04/home/pingu/proj/geoprojects
[TRACE] 17:38:51.812 shell.go:RunCommand(wslpath -m /home/pingu/proj/geoprojects) - 1.533454ms
[TRACE] 17:38:51.812 shell.go:GOOS() - 60ns
[DEBUG] 17:38:51.813 shell.go:CommandPath:597 → /usr/bin/git
[TRACE] 17:38:51.813 shell.go:CommandPath(git) - 958.683µs
[TRACE] 17:38:51.813 shell.go:HasCommand(git) - 962.861µs
[TRACE] 17:38:51.813 shell.go:Pwd() - 90ns
[TRACE] 17:38:51.813 shell.go:HasParentFilePath(.git) - 5.891µs
[DEBUG] 17:38:51.813 properties.go:GetStringArray:52 → exclude_folders: []
[TRACE] 17:38:51.813 shell.go:GOOS() - 40ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 40ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 80ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[DEBUG] 17:38:51.813 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.813 shell.go:Getenv(POSH_GIT_STATUS) - 2.875µs
[DEBUG] 17:38:51.813 properties.go:GetBool:22 → fetch_status: true
[DEBUG] 17:38:51.813 properties.go:GetStringArray:52 → ignore_status: []
[TRACE] 17:38:51.813 shell.go:GOOS() - 30ns
[DEBUG] 17:38:51.813 properties.go:GetKeyValueMap:46 → untracked_modes: map[]
[DEBUG] 17:38:51.813 properties.go:GetKeyValueMap:46 → ignore_submodules: map[]
[DEBUG] 17:38:51.814 shell.go:RunCommand:576 ↓
    # branch.oid 69cce23aaf454567e8501c11defad3e01841ac62
    # branch.head main
    # branch.upstream origin/main
    # branch.ab +0 -0
[TRACE] 17:38:51.814 shell.go:RunCommand(git -C /home/pingu/proj/geoprojects --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 1.024505ms
[DEBUG] 17:38:51.814 properties.go:GetString:28 → NO DATA
[DEBUG] 17:38:51.814 properties.go:GetInt:40 → branch_max_length: 0
[DEBUG] 17:38:51.814 shell.go:HasFolder:490 → false
[TRACE] 17:38:51.814 shell.go:HasFolder(/home/pingu/proj/geoprojects/.git/rebase-merge) - 4.558µs
[DEBUG] 17:38:51.814 shell.go:HasFolder:490 → false
[TRACE] 17:38:51.814 shell.go:HasFolder(/home/pingu/proj/geoprojects/.git/rebase-apply) - 2.896µs
[DEBUG] 17:38:51.814 properties.go:GetString:28 → 
[DEBUG] 17:38:51.814 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.814 shell.go:HasFilesInDir(MERGE_MSG) - 4.459µs
[DEBUG] 17:38:51.814 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.814 shell.go:HasFilesInDir(CHERRY_PICK_HEAD) - 34.776µs
[DEBUG] 17:38:51.814 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.814 shell.go:HasFilesInDir(REVERT_HEAD) - 3.968µs
[DEBUG] 17:38:51.814 shell.go:HasFilesInDir:461 → false
[TRACE] 17:38:51.814 shell.go:HasFilesInDir(sequencer/todo) - 3.166µs
[DEBUG] 17:38:51.814 properties.go:GetString:28 → ≡
[DEBUG] 17:38:51.814 properties.go:GetBool:22 → fetch_upstream_icon: true
[DEBUG] 17:38:51.814 properties.go:GetKeyValueMap:46 → upstream_icons: map[]
[DEBUG] 17:38:51.814 properties.go:GetString:28 → 
[TRACE] 17:38:51.814 shell.go:TemplateCache() - 120ns
[TRACE] 17:38:51.814 shell.go:TemplateCache() - 90ns
[TRACE] 17:38:51.814 shell.go:Shell() - 110ns
[DEBUG] 17:38:51.814 block.go:Debug:230 → Segment: time
[TRACE] 17:38:51.814 shell.go:Pwd() - 100ns
[TRACE] 17:38:51.814 shell.go:GOOS() - 50ns
[DEBUG] 17:38:51.814 properties.go:GetString:28 → 15:04:05
[TRACE] 17:38:51.814 shell.go:TemplateCache() - 60ns
[TRACE] 17:38:51.814 shell.go:TemplateCache() - 40ns
[TRACE] 17:38:51.814 shell.go:Shell() - 100ns
[DEBUG] 17:38:51.814 shell.go:Getenv:395 → NO DATA
[TRACE] 17:38:51.814 shell.go:Getenv(XDG_CACHE_HOME) - 3.036µs
[TRACE] 17:38:51.814 shell_unix.go:CachePath() - 12.063µs
[TRACE] 17:38:51.814 shell.go:Flags() - 50ns

**Reply by RankkaApina** [reply]
Also, as I saw this in another place (but the person didn't call .Venv in the prompt which was their issue), I do see the environment in `$CONDA_DEFAULT_ENV`, but if I run `conda env config vars list`, it's empty.

**Reply by JanDeDobbeleer** [reply]
@RankkaApina your environment variables aren't updated:

```
[DEBUG] 17:38:51.807 shell.go:Getenv:395 → /home/pingu/.pyenv/versions/miniforge3-latest
[TRACE] 17:38:51.807 shell.go:Getenv(VIRTUAL_ENV) - 4.899µs
```

When activating a virtual env, the Python tool should also update the environment variables. That said, I would also update oh-my-posh, as you're on an older version.

Regardless, this isn't something we can solve on our end, you'll need to have a look as to why the VIRTUAL_ENV variable isn't updated when activating the virtual env. Because that should happen.

**Reply by RankkaApina** [reply]
Well, I looked into this, and it seems it's a conda choice to leave VIRTUAL_ENV to pip and only set CONDA_DEFAULT_ENV. I should be able to set this manually though so that it's set for the environment, but it seems the conda is doing what it's supposed to do.

**Reply by JanDeDobbeleer** [reply]
@RankkaApina still, as VIRTUAL_ENV takes precedence over CONDA, that's the one we use. This indeed never works for nested envs, but this is the first time in 6 years I come across that use-case. What could help here? Is it always Conda inside Pip, or are there also cases of Conda, and the pip?

**Reply by RankkaApina** [reply]
So, I tried setting this manually, but that didn't really work... the issue is that the $VIRTUAL_ENV is actually populated by pyenv: /home/pingu/.pyenv/versions/miniforge3-latest
which makes sense: as the virtual environment shows miniforge3-latest, and I think if I mess with that, then the environment won't work correctly, as it needs to know where the pyenv version exists.
So, I guess this would work only if I could separately call the $CONDA_DEFAULT_ENV -> can I? I didn't see this option (and based on your previous answer I don't think it's possible at least currently).

I think I might be a niche case, so I get if this is not something that is added, but I would love to have it... and I'm lazy to learn to code the full prompt by myself :p

**Reply by JanDeDobbeleer** [reply]
@RankkaApina actually, you can do that yourself using the templating engine: `{{ if .Env.CONDA_DEFAULT_ENV }}.Env.CONDA_DEFAULT_ENV{{ else if .Venv }}.Venv{{ end }}`

**Reply by RankkaApina** [reply]
Perfect! That should solve my issue and not mess up my configuration! Thank you :)
